pkg/engine/runtime/terraform/tfops/workspace.go

```diff
@@ -148,6 +149,23 @@ func (w *WorkSpace) InitWorkSpace(ctx context.Context) error {
 
 // Apply with the terraform cli apply command
 func (w *WorkSpace) Apply(ctx context.Context) (*TFState, error) {
+	isHashUpdate := w.checkHashUpdate(ctx)
```

Comment:
Maybe you can extract these codes as a function since there are 16 lines duplicated codes